src/code/z_actor.c

```diff
@@ -2707,19 +2708,88 @@ void Actor_DrawLensActors(PlayState* play, s32 numInvisibleActors, Actor** invis
     CLOSE_DISPS(gfxCtx, "../z_actor.c", 6284);
 }
 
-s32 func_800314B0(PlayState* play, Actor* actor) {
-    return func_800314D4(play, actor, &actor->projectedPos, actor->projectedW);
+/**
+ * Checks if an actor should be culled or not, by seeing if it is contained within it's own culling volume.
+ * For more details on the culling test, see `Actor_CullingVolumeTest`.
+ *
+ * "Culling" in this context refers to the removal of something for the sake of improving performance.
+ * For actors, being culled means that their Update and Draw processes are halted.
+ * While halted, the Actor's update state is frozen and they will not draw, making them invisible.
+ *
+ * Actors that are in bounds of it's culling volume may update and draw, while actors that are
+ * out of bounds of it's culling volume may be excluded from updating and drawing until they are within bounds.
+ *
+ * It is possible for actors to opt out of update culling or draw culling.
+ * This is set per-actor with `ACTOR_FLAG_UPDATE_CULLING_DISABLED` and `ACTOR_FLAG_DRAW_CULLING_DISABLED`.
+ *
+ * Note: Even if either `ACTOR_FLAG_UPDATE_CULLING_DISABLED` or `ACTOR_FLAG_DRAW_CULLING_DISABLED` are set, the actor
+ * will still undergo the culling test and set `ACTOR_FLAG_INSIDE_CULLING_VOLUME` accordingly.
+ * So, `ACTOR_FLAG_INSIDE_CULLING_VOLUME` cannot be used on it own to determine if an actor is actually culled.
+ * It simply says whether or not they are physically located within the bounds of the culling volume.
+ */
+s32 Actor_CullingCheck(PlayState* play, Actor* actor) {
+    return Actor_CullingVolumeTest(play, actor, &actor->projectedPos, actor->projectedW);
 }
 
-s32 func_800314D4(PlayState* play, Actor* actor, Vec3f* arg2, f32 arg3) {
-    f32 var;
+/**
+ * Tests if an actor is currently within the bounds of it's own culling volume.
+ *
+ * The culling volume is a 3D shape composed of a frustum with a box attached to the end of it. The frustum sits at the
+ * camera's position and projects forward, encompassing the player's current view; the box extrudes behind the player,
+ * allowing actors in the immediate vicinty behind and to the sides of the player to be detected.
```

Comment:
```suggestion
 * allowing actors in the immediate vicinity behind and to the sides of the camera to be detected.
```

Reply:
I think its okay to use player here. I changed the first one, because yeah the box only extends behind the camera. But the purpose is to let things be detected nearby the player

Reply:
(I meant to post under this one)
Nevermind, changing it in all places.